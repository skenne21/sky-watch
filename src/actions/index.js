export const addCompanyBio = (bio) => ({
  type: 'ADD_COMPANY_BIO',
  bio
})

export const companyError = (message) => {
  type: 'ADD_ERROR_BIO',
  message
}