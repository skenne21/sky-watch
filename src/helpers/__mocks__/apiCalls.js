import * as mocks from '../../mockData';

export const fetchBio = jest.fn()
  .mockImplementationOnce(() => (mocks.cleanBio))
  .mockImplementationOnce(() => ({
    error: "An error happened"
  }));

export const fetchSpaceXVideos = jest.fn()
  .mockImplementationOnce(() => (mocks.video))
  .mockImplementationOnce(() => ({
    error: "An error happened"
  }));

export const fetchRockets = jest.fn()
  .mockImplementationOnce(() => (mocks.rockets))
  .mockImplementationOnce(() => ({
    error: "An error happened"
  }));

export const fetchCapsules = jest.fn()
  .mockImplementationOnce(() => (mocks.capsules))
  .mockImplementationOnce(() => ({
    error: "An error happened"
  }));

export const fetchLaunchpads = jest.fn()
  .mockImplementationOnce(() => (mocks.launchpads))
  .mockImplementationOnce(() => ({
    error: "An error happened"
  }));

export const fetchLaunches = jest.fn()
  .mockImplementationOnce(() => (mocks.launches))
  .mockImplementationOnce(() => ({
    error: "An error happened"
  }));