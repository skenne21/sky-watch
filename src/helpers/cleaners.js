// tests not passing!!!! Yay!!!!JOY!!!! :(, such a sad face; 

export const cleanVideos = async (videoInfo) => {
  const videos = videoInfo.map(video => ({
    id: video.id.videoId,
    title: video.snippet.title,
    description: video.snippet.description,
    date: video.snippet.publishedAt,
    image: video.snippet.thumbnails.high
  }));
  return videos;
}

export const cleanBios = async (bioInfo) => {
  return {
    name: bioInfo.name,
    founder: bioInfo.founder,
    founded: bioInfo.founded,
    summary: bioInfo.summary,
    headquarters: bioInfo.headquarters
  }
}


export const cleanRockets = (rockets) => {
  return rockets.map(rocket => ({
    name: rocket.name,
    active: rocket.active,
    fristFight: cleanDate(rocket.first_flight),
    costLaunch: rocket.cost_per_launch,
    // stages: rocket.stages,
    country: rocket.country,
    summary: rocket.description,
    stats: {
      diameter: rocket.diameter.feet,
      // engines: cleanEngines(rocket.engines),
      mass: rocket.mass.lb,
      // payload: rocket.payload_weights[0],
      height: rocket.height.feet,
      boosters: rocket.boosters,
      // secondStage: rocket.second_stage,
    } 
  }));
}


export const cleanDate = (date) => {
  const spiltDate = date.split('-');
  const year = spiltDate.shift();
  spiltDate.push(year);
  const recombined = spiltDate.join(', ')
  return recombined;
}

export const cleanEngines = engine => {
  return {
    type: engine.type,
    propellant1: engine.propellant_1,
    propellant2: engine.propellant_2,
    thrustAtSea: engine.thrust_sea_level.lbf,
    thrustVaccum: engine.thrust_vacuum.lbf,
  }
}

export const cleanCapsules = capsules => {
  return capsules.map( capsule => ({
    name: capsule.name,
    crew: capsule.crew_capacity,
    yearInOrbit: capsule.orbit_duration_yr,
    diameter: capsule.diameter.feet,
    heatShield: capsule.heat_shield.temp_degrees,
    height: capsule.height_w_trunk.feet,
    mass: capsule.launch_payload_mass.kg,
  }));
}