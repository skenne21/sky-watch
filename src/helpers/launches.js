import * as cleaners from './cleaners';

export const manageLaunches = launches => {
  const infoFalcon1 = cleaners.filterLaunches(launches, 'falcon1');
  const infoFalcon9 = cleaners.filterLaunches(launches, 'falcon9');
  const infoFalconHeavy = cleaners.filterLaunches(launches, 'falconheavy');
  const falcon1 =  cleaners.cleanLaunches(infoFalcon1);
  const falcon9 =  cleaners.cleanLaunches(infoFalcon9);
  const falconHeavy =  cleaners.cleanLaunches(infoFalconHeavy);
  return cleaners.combineLaunches(falcon1, falcon9, falconHeavy);
};