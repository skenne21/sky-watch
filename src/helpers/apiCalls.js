import * as apiKeys from '../apiKeys';
import * as cleaners from './cleaners';
import { manageLaunches } from './launches';

export const fetchBio = async () => {
  try {
    const response = await fetch(`https://api.spacexdata.com/v2/info`);
    const info = await response.json();
    const bio = await cleaners.cleanBios(info);
    return bio;
  } catch (error) {
    throw error;
  }
};

export const fetchSpaceXVideos = async () => {
  try {
    const baseUrl = 'https://www.googleapis.com/youtube/v3/search';
    const condtions = 'part=snippet&order=viewCount';
    const videoInfo = 'q=spaceX+launches&type=video&videoDefinition=high';
    const key = apiKeys.youTubeKey;
    const response = await fetch(
      `${baseUrl}?${condtions}&${videoInfo}&key=${key}`
    );
    const info = await response.json();
    const videos = await cleaners.cleanVideos(info.items);
    return videos;
  } catch (error) {
    throw error;
  }
};

export const fetchRockets = async () => {
  try {
    const response = await fetch('https://api.spacexdata.com/v2/rockets');
    const info = await response.json();
    const rockets = await cleaners.cleanRockets(info);
    return rockets;
  } catch (error) {
    throw error;
  }
};

export const fetchCapsules = async () => {
  try {
    const response = await fetch('https://api.spacexdata.com/v2/capsules');
    const info = await response.json();
    const capsules = await cleaners.cleanCapsules(info);
    return capsules;
  } catch (error) {
    throw error;
  }
};

export const fetchLaunchpads = async () => {
  try {
    const response = await fetch('https://api.spacexdata.com/v2/launchpads');
    const info = await response.json();
    const launchpads = await cleaners.cleanLaunchpads(info);
    return launchpads;
  } catch (error) {
    throw error;
  }
};

export const fetchLaunches = async () => {
  try {
    const response = await fetch('https://api.spacexdata.com/v2/launches/all');
    const info = await response.json();
    const launches = manageLaunches(info);
    return launches;
  } catch (error) {
    throw error;
  }
};

