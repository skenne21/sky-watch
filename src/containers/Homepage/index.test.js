import React from 'react';
import { shallow } from 'enzyme';
import * as apiCalls from '../../helpers/apiCalls';
import * as actions from '../../actions';
import { mapStateToProps, mapDispatchToProps } from './index';
import { Homepage }from './index';
import * as mocks from '../../mockData';
import { 
  fetchRockets,
  fetchCapsules,
  fetchLaunchpads } from '../../helpers/apiCalls';

jest.mock('../../helpers/apiCalls');

describe('Homepage Container', () => {
  let wrapper;
  let getRockets;
  let getCapsules;
  let getLaunchPads;
  let handleError;
  let companyBio;
  let missionVideos;
  let capsules;
  let launchpads;

  beforeEach(() => {
    getRockets = jest.fn();
    getCapsules = jest.fn();
    getLaunchPads = jest.fn();
    handleError=  jest.fn();
    companyBio=[];
    missionVideos =[];
    capsules= [];
    launchpads= [];

    wrapper = shallow(
      <Homepage
        getRockets={getRockets}
        getCapsules={getCapsules}
        getLaunchPads={getLaunchPads}
        handleError={handleError}
        companyBio={companyBio}
        missionVideos={missionVideos}
        capsules={capsules}
        launchpads={launchpads} 
      />
    );
  });

  it('Should match the snapshot if the bio has length', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Should Match the snapshot if there is no length to the bio', () => {
    wrapper = shallow(
      <Homepage
        getRockets={getRockets}
        getCapsules={getCapsules}
        getLaunchPads={getLaunchPads}
        handleError={handleError}
        companyBio={[]}
        missionVideos={missionVideos}
        capsules={capsules}
        launchpads={launchpads} 
      />
    );
    expect(wrapper).toMatchSnapshot();
  })

  it('Should call handleRockets when clicked', () => {
    const navlink = wrapper.find('.rockets').simulate('click')
    expect(fetchRockets).toHaveBeenCalled();
  });

  it('Should call handleCapsules when clicked', () => {
    const navlink = wrapper.find('.capsules').simulate('click')
    expect(fetchCapsules).toHaveBeenCalled();
  });

  it('Should call handleLaunchpads when clicked', () => {
    const navlink = wrapper.find('.launchpad').simulate('click')
    expect(fetchLaunchpads).toHaveBeenCalled();
  });

  it('Should call handleLaunches when clicked', () => {
    const navlink = wrapper.find('.launches').simulate('click')
    expect(fetchLaunchpads).toHaveBeenCalled();
  });

  describe('mapStateToProps', () => {
    let mockedState, expectedState;
    beforeEach(() => {
      mockedState = {
        companyBio: [],
        missionVideos: [],
        rockets: [],
        capsules: [],
        launchpads: [],
        user: {}
      }

      expectedState = {
        companyBio: [],
        missionVideos: [],
        rockets: [],
        capsules: [],
        launchpads: [],
        user: {}
      }
    })

    it('should map companyBio to state', () => {
      const bio = mocks.bio;
      mockedState.companyBio = [bio];
      const mapped = mapStateToProps(mockedState);
      expectedState.companyBio = [bio];
      expect(mapped).toEqual(expectedState);
    });

    it('Should map missionVideos to state', () => {
      const videos = mocks.videos;
      mockedState.missionVideos = videos;
      const mapped = mapStateToProps(mockedState);
      expectedState.missionVideos = videos;
      expect(mapped).toEqual(expectedState);
    });

    it('Should map Rockets to state', () => {
      const rockets = mocks.rockets;
      mockedState.rockets = rockets;
      const mapped = mapStateToProps(mockedState);
      expectedState.rockets = rockets;
      expect(mapped).toEqual(expectedState);
    });

    it('Should map capsules to state' , () => {
      const capsules = mocks.capsules;
      mockedState.capsules = capsules;
      const mapped = mapStateToProps(mockedState);
      expectedState.capsules = capsules;
      expect(mapped).toEqual(expectedState);
    });

    it('Should map launchpads to state', () => {
      const launchpads = mocks.launchpads;
      mockedState.launchpads = launchpads;
      const mapped = mapStateToProps(mockedState);
      expectedState.launchpads = launchpads;
      expect(mapped).toEqual(expectedState);
    });

    it('Should map user to state', () => {
      const user = {
        name:'bob',
        id: 1,
        email:'bob@gmail.com'
      };

      mockedState.user = user;
      const mapped = mapStateToProps(mockedState);
      expectedState.user = user;
      expect(mapped).toEqual(expectedState);
    });
  });

  describe('MapDispatchToProps', () => {
    let mockedDispatch, mapped;

    beforeEach(() => {
      mockedDispatch = jest.fn();
      mapped = mapDispatchToProps(mockedDispatch);
    })

    it('Should dispatch getRockets with the right params', () => {
      const rockets = mocks.rockets;
      mapped.getRockets(rockets);
      const expected = actions.addRockets(rockets);

      expect(mockedDispatch).toHaveBeenCalledWith(expected);
    });

    it('Should dispatch getCapsules with the right params', () => {
      const capsules = mocks.capsules;
      mapped.getCapsules(capsules);

      const expected = actions.addCapsules(capsules);

      expect(mockedDispatch).toHaveBeenCalledWith(expected);
    });

    it('Should dispatch getLaunchPads with the right params', () => {
      const launchpads = mocks.launchpads;
      mapped.getLaunchpads(launchpads);
      const expected = actions.addLaunchPads(launchpads);

      expect(mockedDispatch).toHaveBeenCalledWith(expected);
    });

    it('Should dispatch handleError with the right params', ( ) => {
      const error = { message: 'error'};
      mapped.handleError(error.message);

      const expected = actions.addError(error.message);
      expect(mockedDispatch).toHaveBeenCalledWith(expected);
    })
  });
});