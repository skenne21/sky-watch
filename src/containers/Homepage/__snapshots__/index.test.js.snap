// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Container Should Match the snapshot if there is no length to the bio 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Homepage
    capsules={Array []}
    companyBio={Array []}
    getCapsules={[Function]}
    getLaunchPads={[Function]}
    getRockets={[Function]}
    handleError={[Function]}
    launchpads={Array []}
    missionVideos={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="loading"
>
          <img
                    alt="Picture of starman"
                    src="../../assets/starman.png"
          />
          <p>
                     Loading 
          </p>
</div>,
        <form
          className="nav-links"
>
          <NavLink
                    activeClassName="isActive"
                    ariaCurrent="true"
                    className="rockets"
                    name="rockets"
                    onClick={[Function]}
                    to="/rockets"
          >
                    ROCKETS
          </NavLink>
          <NavLink
                    activeClassName="isActive"
                    ariaCurrent="true"
                    className="capsules"
                    name="capsules"
                    onClick={[Function]}
                    to="/capsules"
          >
                    CAPSULES
          </NavLink>
          <NavLink
                    activeClassName="isActive"
                    ariaCurrent="true"
                    className="launchpad"
                    name="launchpad"
                    onClick={[Function]}
                    to="/launchpads"
          >
                    LAUNCHPADS
          </NavLink>
          <NavLink
                    activeClassName="isActive"
                    ariaCurrent="true"
                    className="launches"
                    name="launches"
                    onClick={[Function]}
                    to="/launches"
          >
                    LAUNCHES
          </NavLink>
</form>,
      ],
      "className": "Homepage",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Picture of starman"
              src="../../assets/starman.png"
/>,
            <p>
               Loading 
</p>,
          ],
          "className": "loading",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Picture of starman",
              "src": "../../assets/starman.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Loading ",
            },
            "ref": null,
            "rendered": " Loading ",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="isActive"
              ariaCurrent="true"
              className="rockets"
              name="rockets"
              onClick={[Function]}
              to="/rockets"
>
              ROCKETS
</NavLink>,
            <NavLink
              activeClassName="isActive"
              ariaCurrent="true"
              className="capsules"
              name="capsules"
              onClick={[Function]}
              to="/capsules"
>
              CAPSULES
</NavLink>,
            <NavLink
              activeClassName="isActive"
              ariaCurrent="true"
              className="launchpad"
              name="launchpad"
              onClick={[Function]}
              to="/launchpads"
>
              LAUNCHPADS
</NavLink>,
            <NavLink
              activeClassName="isActive"
              ariaCurrent="true"
              className="launches"
              name="launches"
              onClick={[Function]}
              to="/launches"
>
              LAUNCHES
</NavLink>,
          ],
          "className": "nav-links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "isActive",
              "ariaCurrent": "true",
              "children": "ROCKETS",
              "className": "rockets",
              "name": "rockets",
              "onClick": [Function],
              "to": "/rockets",
            },
            "ref": null,
            "rendered": "ROCKETS",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "isActive",
              "ariaCurrent": "true",
              "children": "CAPSULES",
              "className": "capsules",
              "name": "capsules",
              "onClick": [Function],
              "to": "/capsules",
            },
            "ref": null,
            "rendered": "CAPSULES",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "isActive",
              "ariaCurrent": "true",
              "children": "LAUNCHPADS",
              "className": "launchpad",
              "name": "launchpad",
              "onClick": [Function],
              "to": "/launchpads",
            },
            "ref": null,
            "rendered": "LAUNCHPADS",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "isActive",
              "ariaCurrent": "true",
              "children": "LAUNCHES",
              "className": "launches",
              "name": "launches",
              "onClick": [Function],
              "to": "/launches",
            },
            "ref": null,
            "rendered": "LAUNCHES",
            "type": [Function],
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="loading"
>
            <img
                        alt="Picture of starman"
                        src="../../assets/starman.png"
            />
            <p>
                         Loading 
            </p>
</div>,
          <form
            className="nav-links"
>
            <NavLink
                        activeClassName="isActive"
                        ariaCurrent="true"
                        className="rockets"
                        name="rockets"
                        onClick={[Function]}
                        to="/rockets"
            >
                        ROCKETS
            </NavLink>
            <NavLink
                        activeClassName="isActive"
                        ariaCurrent="true"
                        className="capsules"
                        name="capsules"
                        onClick={[Function]}
                        to="/capsules"
            >
                        CAPSULES
            </NavLink>
            <NavLink
                        activeClassName="isActive"
                        ariaCurrent="true"
                        className="launchpad"
                        name="launchpad"
                        onClick={[Function]}
                        to="/launchpads"
            >
                        LAUNCHPADS
            </NavLink>
            <NavLink
                        activeClassName="isActive"
                        ariaCurrent="true"
                        className="launches"
                        name="launches"
                        onClick={[Function]}
                        to="/launches"
            >
                        LAUNCHES
            </NavLink>
</form>,
        ],
        "className": "Homepage",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Picture of starman"
                src="../../assets/starman.png"
/>,
              <p>
                 Loading 
</p>,
            ],
            "className": "loading",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Picture of starman",
                "src": "../../assets/starman.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Loading ",
              },
              "ref": null,
              "rendered": " Loading ",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="isActive"
                ariaCurrent="true"
                className="rockets"
                name="rockets"
                onClick={[Function]}
                to="/rockets"
>
                ROCKETS
</NavLink>,
              <NavLink
                activeClassName="isActive"
                ariaCurrent="true"
                className="capsules"
                name="capsules"
                onClick={[Function]}
                to="/capsules"
>
                CAPSULES
</NavLink>,
              <NavLink
                activeClassName="isActive"
                ariaCurrent="true"
                className="launchpad"
                name="launchpad"
                onClick={[Function]}
                to="/launchpads"
>
                LAUNCHPADS
</NavLink>,
              <NavLink
                activeClassName="isActive"
                ariaCurrent="true"
                className="launches"
                name="launches"
                onClick={[Function]}
                to="/launches"
>
                LAUNCHES
</NavLink>,
            ],
            "className": "nav-links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "ariaCurrent": "true",
                "children": "ROCKETS",
                "className": "rockets",
                "name": "rockets",
                "onClick": [Function],
                "to": "/rockets",
              },
              "ref": null,
              "rendered": "ROCKETS",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "ariaCurrent": "true",
                "children": "CAPSULES",
                "className": "capsules",
                "name": "capsules",
                "onClick": [Function],
                "to": "/capsules",
              },
              "ref": null,
              "rendered": "CAPSULES",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "ariaCurrent": "true",
                "children": "LAUNCHPADS",
                "className": "launchpad",
                "name": "launchpad",
                "onClick": [Function],
                "to": "/launchpads",
              },
              "ref": null,
              "rendered": "LAUNCHPADS",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "ariaCurrent": "true",
                "children": "LAUNCHES",
                "className": "launches",
                "name": "launches",
                "onClick": [Function],
                "to": "/launches",
              },
              "ref": null,
              "rendered": "LAUNCHES",
              "type": [Function],
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Homepage Container Should match the snapshot if the bio has length 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Homepage
    capsules={Array []}
    companyBio={Array []}
    getCapsules={[Function]}
    getLaunchPads={[Function]}
    getRockets={[Function]}
    handleError={[Function]}
    launchpads={Array []}
    missionVideos={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="loading"
>
          <img
                    alt="Picture of starman"
                    src="../../assets/starman.png"
          />
          <p>
                     Loading 
          </p>
</div>,
        <form
          className="nav-links"
>
          <NavLink
                    activeClassName="isActive"
                    ariaCurrent="true"
                    className="rockets"
                    name="rockets"
                    onClick={[Function]}
                    to="/rockets"
          >
                    ROCKETS
          </NavLink>
          <NavLink
                    activeClassName="isActive"
                    ariaCurrent="true"
                    className="capsules"
                    name="capsules"
                    onClick={[Function]}
                    to="/capsules"
          >
                    CAPSULES
          </NavLink>
          <NavLink
                    activeClassName="isActive"
                    ariaCurrent="true"
                    className="launchpad"
                    name="launchpad"
                    onClick={[Function]}
                    to="/launchpads"
          >
                    LAUNCHPADS
          </NavLink>
          <NavLink
                    activeClassName="isActive"
                    ariaCurrent="true"
                    className="launches"
                    name="launches"
                    onClick={[Function]}
                    to="/launches"
          >
                    LAUNCHES
          </NavLink>
</form>,
      ],
      "className": "Homepage",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Picture of starman"
              src="../../assets/starman.png"
/>,
            <p>
               Loading 
</p>,
          ],
          "className": "loading",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Picture of starman",
              "src": "../../assets/starman.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Loading ",
            },
            "ref": null,
            "rendered": " Loading ",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="isActive"
              ariaCurrent="true"
              className="rockets"
              name="rockets"
              onClick={[Function]}
              to="/rockets"
>
              ROCKETS
</NavLink>,
            <NavLink
              activeClassName="isActive"
              ariaCurrent="true"
              className="capsules"
              name="capsules"
              onClick={[Function]}
              to="/capsules"
>
              CAPSULES
</NavLink>,
            <NavLink
              activeClassName="isActive"
              ariaCurrent="true"
              className="launchpad"
              name="launchpad"
              onClick={[Function]}
              to="/launchpads"
>
              LAUNCHPADS
</NavLink>,
            <NavLink
              activeClassName="isActive"
              ariaCurrent="true"
              className="launches"
              name="launches"
              onClick={[Function]}
              to="/launches"
>
              LAUNCHES
</NavLink>,
          ],
          "className": "nav-links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "isActive",
              "ariaCurrent": "true",
              "children": "ROCKETS",
              "className": "rockets",
              "name": "rockets",
              "onClick": [Function],
              "to": "/rockets",
            },
            "ref": null,
            "rendered": "ROCKETS",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "isActive",
              "ariaCurrent": "true",
              "children": "CAPSULES",
              "className": "capsules",
              "name": "capsules",
              "onClick": [Function],
              "to": "/capsules",
            },
            "ref": null,
            "rendered": "CAPSULES",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "isActive",
              "ariaCurrent": "true",
              "children": "LAUNCHPADS",
              "className": "launchpad",
              "name": "launchpad",
              "onClick": [Function],
              "to": "/launchpads",
            },
            "ref": null,
            "rendered": "LAUNCHPADS",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "isActive",
              "ariaCurrent": "true",
              "children": "LAUNCHES",
              "className": "launches",
              "name": "launches",
              "onClick": [Function],
              "to": "/launches",
            },
            "ref": null,
            "rendered": "LAUNCHES",
            "type": [Function],
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="loading"
>
            <img
                        alt="Picture of starman"
                        src="../../assets/starman.png"
            />
            <p>
                         Loading 
            </p>
</div>,
          <form
            className="nav-links"
>
            <NavLink
                        activeClassName="isActive"
                        ariaCurrent="true"
                        className="rockets"
                        name="rockets"
                        onClick={[Function]}
                        to="/rockets"
            >
                        ROCKETS
            </NavLink>
            <NavLink
                        activeClassName="isActive"
                        ariaCurrent="true"
                        className="capsules"
                        name="capsules"
                        onClick={[Function]}
                        to="/capsules"
            >
                        CAPSULES
            </NavLink>
            <NavLink
                        activeClassName="isActive"
                        ariaCurrent="true"
                        className="launchpad"
                        name="launchpad"
                        onClick={[Function]}
                        to="/launchpads"
            >
                        LAUNCHPADS
            </NavLink>
            <NavLink
                        activeClassName="isActive"
                        ariaCurrent="true"
                        className="launches"
                        name="launches"
                        onClick={[Function]}
                        to="/launches"
            >
                        LAUNCHES
            </NavLink>
</form>,
        ],
        "className": "Homepage",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Picture of starman"
                src="../../assets/starman.png"
/>,
              <p>
                 Loading 
</p>,
            ],
            "className": "loading",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Picture of starman",
                "src": "../../assets/starman.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Loading ",
              },
              "ref": null,
              "rendered": " Loading ",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="isActive"
                ariaCurrent="true"
                className="rockets"
                name="rockets"
                onClick={[Function]}
                to="/rockets"
>
                ROCKETS
</NavLink>,
              <NavLink
                activeClassName="isActive"
                ariaCurrent="true"
                className="capsules"
                name="capsules"
                onClick={[Function]}
                to="/capsules"
>
                CAPSULES
</NavLink>,
              <NavLink
                activeClassName="isActive"
                ariaCurrent="true"
                className="launchpad"
                name="launchpad"
                onClick={[Function]}
                to="/launchpads"
>
                LAUNCHPADS
</NavLink>,
              <NavLink
                activeClassName="isActive"
                ariaCurrent="true"
                className="launches"
                name="launches"
                onClick={[Function]}
                to="/launches"
>
                LAUNCHES
</NavLink>,
            ],
            "className": "nav-links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "ariaCurrent": "true",
                "children": "ROCKETS",
                "className": "rockets",
                "name": "rockets",
                "onClick": [Function],
                "to": "/rockets",
              },
              "ref": null,
              "rendered": "ROCKETS",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "ariaCurrent": "true",
                "children": "CAPSULES",
                "className": "capsules",
                "name": "capsules",
                "onClick": [Function],
                "to": "/capsules",
              },
              "ref": null,
              "rendered": "CAPSULES",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "ariaCurrent": "true",
                "children": "LAUNCHPADS",
                "className": "launchpad",
                "name": "launchpad",
                "onClick": [Function],
                "to": "/launchpads",
              },
              "ref": null,
              "rendered": "LAUNCHPADS",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "ariaCurrent": "true",
                "children": "LAUNCHES",
                "className": "launches",
                "name": "launches",
                "onClick": [Function],
                "to": "/launches",
              },
              "ref": null,
              "rendered": "LAUNCHES",
              "type": [Function],
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
